url969d1ddca4_compmstr_java_scheme_tgz-pJ8-com_undi_javascheme_JavaSchemeJ8/src/com/undi/javascheme/SchemeEval.java

```diff
@@ -0,0 +1,804 @@
+      /*if(frameMap.containsKey(var)){
+        return frameMap.get(var);
+      }*/
+      result = frameMap.get(var);
+      if(result != null){
+        return result;
+      }
+      env = enclosingEnvironment(env);
+    }
+    //Check if this is a java-defined name
+    //Check for instance/static fields
+    if(var.getSymbol().contains("/")){
+      String[] parts = var.getSymbol().split("/");
+      if(parts.length == 2){
+        String varName = parts[0];
+        String fieldName = parts[1];
+        //Check if varName is a variable
+        try{
+          SchemeObject javaObj = lookupVariableValue(SchemeObject.makeSymbol(varName), originalEnv);
+          return SchemeObject.makeJavaObj(Reflector.getInstanceField(javaObj.getJavaObj(), fieldName));
+        }catch(IllegalArgumentException ex){
+          //Not a variable, try for static field
+          try{
+            return SchemeObject.makeJavaObj(Reflector.getStaticField(varName, fieldName));
+          }catch(Exception e){
+            //Not a field, try for a static method 
+            return SchemeObject.makeJavaStaticMethod(varName, fieldName);
+          }
+        }
+      }
+    }
+    //Check for classes/methods
+    if(var.getSymbol().startsWith(".") && var.getSymbol().length() > 1){
+      System.out.println("Java method/member: " + var.getSymbol().substring(1));
+      return SchemeObject.makeJavaMethod(var.getSymbol().substring(1));
+    }else if(var.getSymbol().endsWith(".")){
+      String className = var.getSymbol().substring(0, var.getSymbol().length() - 1);
+      System.out.println("Java Constructor: " + className);
+      return SchemeObject.makeJavaConstructor(className);
+    }
+    try{
+    	Class cls = Reflector.classFromName(var.getSymbol());
+    	if(cls != null){
+    		return SchemeObject.makeJavaObj(cls);
+    	}
+    }catch(Exception e){ }
+    throw new IllegalArgumentException("Unbound Variable: " + var.getSymbol());
+  }
+  
+  public void setVariableValue(SchemeObject var, SchemeObject val, SchemeObject env){
+    HashMap<SchemeObject, SchemeObject> frameMap;
+    while(!env.isEmptyList()){
+      frameMap = firstFrame(env).getHashMap();
+      if(frameMap.containsKey(var)){
+        frameMap.put(var, val);
+        return;
+      }
+      env = enclosingEnvironment(env);
+    }
+		throw new SchemeException("Unbound Variable - " + var.getSymbol());
+  }
+  
+  public void defineVariable(SchemeObject var, SchemeObject val, SchemeObject env){
+    SchemeObject frame;
+    frame = firstFrame(env);
+    addBindingToFrame(var, val, frame);
+  }
+  
+  public SchemeObject setupEnvironment(){
+    SchemeObject initialEnv;
+    initialEnv = extendEnvironment(SchemeObject.THE_EMPTY_LIST,
+                                      SchemeObject.THE_EMPTY_LIST, 
+                                      emptyEnvironment);
+    
+    return initialEnv;
+  }
+  
+  public SchemeObject eval(SchemeObject exp, SchemeObject env){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 38. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.